### ISSUE TITLE
Add topologySpreadConstraints

### ISSUE THREAD

**Issue by RonPortnoy** [kiali/kiali#5614]
Add the option to create kiali in HA. 
spec.topologySpreadConstraints

**Comment by jmazzitelli** [comment]
k8s docs on this setting: https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/

**Comment by jshaughn** [comment]
Closing due to inactivity.  Can be re-opened or replaced with a more recent approach.

**Comment by frittentheke** [comment]
@jshaughn  I was just about to open an issue asking to add configurability for `topologySpreadConstraints`.
Would you possible reopen this one?

In the end this is just like the `spec.deployment.affinity` setting allowing to configure the pod placement.

**Comment by jshaughn** [comment]
Hi @frittentheke , will re-open. @jmazzitelli , any opinion on this?

**Comment by jmazzitelli** [comment]
Should be a simple enhancement. This will require two PRs - one for the operator and one for the server helm chart.

FWIW: here's the original PR that introduced the affinity stuff to the operator - we want something similar: https://github.com/kiali/kiali/pull/1304 .. it was introduced so long that we didn't have the server helm chart at that time. But there is something similar there: https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/deployment.yaml#L219-L233

This would be altering the server resource with a new config option, so the checklists involved would be:

* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-resource
* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-configuration-setting

**Comment by frittentheke** [comment]
> [@frittentheke](https://github.com/frittentheke) Any interest in contributing a PR?

Yes, potentially. I was still trying to grasp what amount of little changes this supposedly simple addition requires.
Thanks to @jmazzitelli for already collecting some reference PRs and issues.

The operator makes this endeavor much more complex than just adding some templating and a value to the Helm chart.
And I suppose all of this would have to come together to makes this a consistent feature.

Not to kick off an unrelated discussion but out of real interest: Are there still strong reasons to have an operator with all of its baggage (CRs,...). Especially considering the Istio Operator is gone ... https://istio.io/latest/blog/2024/in-cluster-operator-deprecation-announcement/ it seems strange "having" (I know there is a documented Helm installation path) an operator for Kiali.

**Comment by jmazzitelli** [comment]
> Are there still strong reasons to have an operator with all of its baggage (CRs,...). Especially considering the Istio Operator is gone 

There are strong reasons, the main one being the Kiali team strongly supports OpenShift, and OpenShift is operator-centric. The Kiali Operator integrates nicely in the OpenShift ecosystem.

And while the "Istio Operator" is gone, it is being replaced with the Sail Operator in the istio-ecosystem - check it out :) https://github.com/istio-ecosystem/sail-operator

**Comment by jmazzitelli** [comment]
My plan was to look into implementing this today myself. It won't take me long to do it. Hopefully, it will be available in our next sprint release to be published this coming Monday. Stay tuned.

**Comment by frittentheke** [comment]
> My plan was to look into implementing this today myself. It won't take me long to do it. Hopefully, it will be available in our next sprint release to be published this coming Monday. Stay tuned.

AWE SOME! Thanks a bunch!

**Comment by jmazzitelli** [comment]
PRs: 
* https://github.com/kiali/kiali-operator/pull/873
* https://github.com/kiali/helm-charts/pull/308

Test procedures are documented in the PRs.
